plugin/evm/limit_order.go

```diff
@@ -72,13 +73,20 @@ func (lop *limitOrderProcesser) RunMatchingEngine() {
 	if len(longOrders) == 0 || len(shortOrders) == 0 {
 		return
 	}
-	for _, longOrder := range longOrders {
-		for j, shortOrder := range shortOrders {
-			if longOrder.Price == shortOrder.Price && longOrder.BaseAssetQuantity == (-shortOrder.BaseAssetQuantity) {
-				err := lop.limitOrderTxProcessor.ExecuteMatchedOrdersTx(*longOrder, *shortOrder)
+	for i := 0; i < len(longOrders); i++ {
+		for j := 0; j < len(shortOrders); j++ {
+			if getUnFilledBaseAssetQuantity(longOrders[i]) == 0 {
```

Comment:
This check can happen before the second loop, with a continue.

Reply:
longOrders = L1 , L2
shortOrders = S1,S2,S3

1st iteration
L1 and S1 exhaust each other
we need to break inner loop so that we dont continue matching L1 with S2 and S3.
So check's need to be inside loop.